pkg/diagnostics/client/client.go

```diff
@@ -0,0 +1,145 @@
+package client
+
+import (
+	"fmt"
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	clientcmdapi "github.com/GoogleCloudPlatform/kubernetes/pkg/client/clientcmd/api"
+	//"github.com/kr/pretty"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+var Diagnostics = map[string]types.Diagnostic{
+	"NodeDefinitions": {
+		Description: "Check node records on master",
+		Condition: func(env *types.Environment) (skip bool, reason string) {
+			if env.ClusterAdminFactory == nil {
+				return true, "Client does not have cluster-admin access and cannot see node records"
+			}
+			return false, ""
+		},
+		Run: func(env *types.Environment) {
+			var err error
+			var nodes *kapi.NodeList
+			if _, kclient, err := env.ClusterAdminFactory.Clients(); err == nil {
+				nodes, err = kclient.Nodes().List(labels.LabelSelector{})
+			}
+			if err != nil {
```

Comment:
As written it also addresses the case where err is set from listing nodes (inside the if)